draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -716,60 +712,36 @@ This section describes how subscribers present and relying parties verify Merkle
 
-2. Check if the certificate's `batch_number` is contained in the saved validity window. If not, abort this procedure with a `unknown_ca` error.
+For a Merkle Tree trust anchor, the trust anchor will identify a batch in the relying party's validity window. (See {{identifying}} and {{relying-parties}}.) The relying party runs the following procedure:
 
-3. Compute the expiration time of the certificate's `batch_number`, as described in {{parameters}}. If this value is before the current time, abort this procedure with a `certificate_expired` error.
+1. Compute the batch's expiration time, as described in {{parameters}}. If this value is before the current time, abort this procedure with a `certificate_expired` error.
 
-4. Set `hash` to the output of `HashAssertion(assertion, index)`. Set `remaining` to the certificate's `index` value.
+2. Set `hash` to the output of `HashAssertion(assertion, index)`. Set `remaining` to the certificate's `index` value.
 
-5. For each element `v` at zero-based index `i` of the certificate's `path` field, in order:
+3. For each element `v` at zero-based index `i` of the certificate's `path` field, in order:
 
    - If `remaining` is odd, set `hash` to the output of `HashNode(v, hash, i + 1, remaining >> 1)`.
      Otherwise, set `hash` to the output of `HashNode(hash, v, i + 1, remaining >> 1)`
 
    - Set `remaining` to `remaining >> 1`.
 
-5. If `remaining` is non-zero, abort this procedure with an error.
-
-6. If `hash` is not equal to the corresponding tree head in the saved validity window, abort this procedure with a `bad_certificate` error.
-
-7. Optionally, perform any additional application-specific checks on the assertion and issuer. For example, an HTTPS client might constrain an issuer to a particular DNS subtree.
-
-8. If all the preceding checks succeed, the certificate is valid and the application can proceed with using the assertion.
-
-## Certificate Negotiation {#trust-anchor-negotiation}
-
-Merkle Tree certificates can only be presented to up-to-date relying parties, so this document describes a mechanism for subscribers to select certificates. This section describes the general negotiation mechanism. {{tls-trust-anchors-extension}} describes it as used in TLS.
+4. If `remaining` is non-zero, abort this procedure with an error.
 
-Subscribers maintain a certificate set of available BikeshedCertificates. The TrustAnchor value in each BikeshedCertificate is known as the primary TrustAnchor. Each BikeshedCertificate is also associated with the following values:
+5. If `hash` is not equal to the batch's tree head in the relying party's saved validity window (see {{relying-parties}}), abort this procedure with a `bad_certificate` error.
 
-* A set of additional TrustAnchor values which also match this certificate
+6. Optionally, perform any additional application-specific checks on the assertion and issuer. For example, an HTTPS client might constrain an issuer to a particular DNS subtree.
```

Comment:
Self-review: this and step 7 should be part of the generic process, not the issuer-specific process.